source/azure_iot_private.h

```diff
@@ -0,0 +1,25 @@
+/* Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License. */
+
+#ifndef AZURE_IOT_PRIVATE_H
+#define AZURE_IOT_PRIVATE_H
+
+#include "azure_iot_result.h"
+
+#include "azure/az_core.h"
+#include "azure/az_iot.h"
+
+#include <azure/core/_az_cfg_prefix.h>
+
+/**
+ * @brief Translate embedded errors to middleware errors
+ *
+ * @param[in] xCoreError The `az_result` to translate to a middleware error
+ *
+ * @result The #AzureIoTResult_t translated from \p xCoreError.
+ */
+AzureIoTResult_t prvAzureIoT_TranslateCoreError( az_result xCoreError );
```

Comment:
>prvAzureIoT_TranslateCoreError [](http://example.com/codeflow?start=17&length=30)

prv is generally used with static function.


"File scope static (private) functions are prefixed with prv."

[https://www.freertos.org/FreeRTOS-Coding-Standard-and-Style-Guide.html](https://www.freertos.org/FreeRTOS-Coding-Standard-and-Style-Guide.html)


How about not adding additional meaning to this prv? Can we name it standard 
xAzureIoT_TranslateCoreError

and let include file name show that it is private ?

Reply:
I'd rather keep the `prv`. The wording there in parenthesis is "private" which I think this function adheres to the essence of the rule. Technically, one could argue it's breaking convention. But fundamentally it would be file specific if it could. Since FreeRTOS doesn't have guidance on this specific scenario, I think it's up to us to decide which camp it belongs in. I think it's closer the `prvAzureIoT` than it is to `xAzureIot`. Plus, since this is internal, it's us that has to deal with the concrete usage and customers would and should only be reading the code. Seeing the keyword `prv` triggers in their head that they should basically ignore.

Reply:
Talked offline with ewerton to get another opinion and he agrees with you. Making the change now

Reply:
Talked offline with ewerton to get another opinion and he agrees with you. Making the change now